opfpush/src/main/java/org/onepf/opfpush/OPFPushHelper.java

```diff
@@ -139,14 +147,16 @@ public void register() {
     }
 
     /**
-     * Unregister the application. Calling unregister() stops any messages from the server.
-     * This is a not blocking call. You should rarely (if ever) need to call this method.
-     * Not only is it expensive in terms of resources, but it invalidates your registration ID,
-     * which you should never change unnecessarily.
+     * If {@code OPFPushHelper} is registered for pushes or registration in process,
+     * starts asynchronous unregistration of current push provider. Does nothing if {@code OPFPushHelper}
+     * already unregistered.
+     * <p/>
+     * You should rarely (if ever) need to call this method. Not only is it expensive in terms of resources,
+     * but it invalidates your registration ID, which you should never change unnecessarily.
```

Comment:
Strange order of words, difficult to understand. Maybe smth like this:
You should rarely (if ever) call this method. Not only because it is expensive in terms of resources, but also because it invalidates your registration ID, which you should not change in normal situation.